site/_posts/2020-09-03-oidc-auth-api-usage.md

```diff
@@ -0,0 +1,173 @@
+author: jvlcek
+date: 2020-09-03
+comments: true
+published: true
+tags: tutorials
+---
+
+The goal of this blog post is to provide a basic understanding of how to use the ManageIQ REST API when configured with OpenID-Connect authentication.
+
+All the content in this blog post is only relevant when ManageIQ is configured for OpenID-Connect Authentication.
+
+**Contents:**
+---------------------------------------------------------------------
+
+- [**Configure ManageIQ For OpenID-Connect Authentication**](#configure-manageiq-for-openid-connect-authentication)
+- [**Using The ManageIQ REST API**](#using-the-manageiq-rest-api)
+- [**The OpenID-Connect REST API Authentication Modes**](#the-openid-connect-rest-api-authentication-modes)
+- [**Examples**](#examples)
+  - [Example: Using a JWT Token](#example:-using-a-jwt-token)
+  - [Example: Using a ManageIQ API Auth Token](#example:-using-a-manageiq-api-auth-token)
+  - [Example: Using Basic Auth For Admin](#example:-using-basic-auth-for-admin)
+
+# Configure ManageIQ For OpenID-Connect Authentication
+---------------------------------------------------------------------
+
+ManageIQ uses Apache’s <em>mod_auth_openidc</em> module to support OpenID-Connect authentication.
+The instructions for configuring Apache’s <em>mod_auth_openidc</em> for ManageIQ	differ for the
+appliance deployment and the podified (Kubernetes) deployment.
+
+  + The instructions for configuring OpenID-Connect for a ManageIQ appliance can be found here:
+    [ManageIQ OpenID-Connect Authentication User Reference](https://www.manageiq.org/docs/reference/latest/auth/openid_connect.html)
+
+  + The instructions for configuring OpenID-Connect for a podifed deployment of ManageIQ can be found here:
+    [ManageIQ Operator README](https://github.com/ManageIQ/manageiq-pods/blob/master/manageiq-operator/README.md#configuring-openid-connect-authentication)
+
+# Using The ManageIQ REST API
+---------------------------------------------------------------------
+
+Please see the [ManageIQ REST API documentation](https://www.manageiq.org/docs/api) for an overview
+and detailed reference.
+
+# The OpenID-Connect REST API Authentication Modes
+---------------------------------------------------------------------
+
+When configured for OpenID-Connect there are three different authentication modes:
+
+  + A Java Web Token (JWT) obtained from the OpenID-Connect provider for a non-admin user.
+  + A ManageIQ API Auth Token for a non-admin user.
+    A JWT is required to obtain the ManageIQ API Auth Token but performance is improved using the ManageIQ API Auth Token for consecutive API requests
+  + Basic authentication, username:password format, is only supported for the *admin*  user.
+
+# Examples
+---------------------------------------------------------------------
+
+The below examples are BASH(1) shell script commands. They assume the
+*JQ(1)*  <em>Command-line JSON processor</em> is available and the following
+variable be defined:
```

Comment:
variables

Reply:
maybe also add, they assume the client created in Keycloak in a realm names `miq`